backend/src/gpml/handler/review.clj

```diff
@@ -1,13 +1,86 @@
-(defmethod ig/init-key :gpml.handler.review/get-reviewers [_ {:keys [db]}]
+(defn get-review [db topic-type topic-id]
+  (let [conn (:spec db)
+        topic-name (cond
+                     (contains? (set constants/resource-types) topic-type) "resource"
+                     (= topic-type "project") "initiative"
+                     :else topic-type)
+        review (db.review/review-by-topic-item
+                conn
+                {:topic-name topic-name :topic-id topic-id})]
+    (resp/response review)))
+
+(defn assign-reviewer [db topic-type topic-id reviewer assigned-by]
+  (let [topic-name (cond
+                     (contains? (set constants/resource-types) topic-type) "resource"
+                     (= topic-type "project") "initiative"
+                     :else topic-type)]
+    (jdbc/with-db-transaction [conn (:spec db)]
+      (let [existing (db.review/review-by-topic-item
+                      conn
+                      {:topic-name topic-name :topic-id topic-id})
+            review-id (:id existing)
+            updated (if (nil? review-id)
+                      (db.review/new-review conn {:topic-name topic-name
+                                                  :topic-id topic-id
+                                                  :assigned-by assigned-by
+                                                  :reviewer reviewer})
+                      (db.review/change-reviewer conn {:id review-id
+                                                       :assigned-by assigned-by
+                                                       :reviewer reviewer}))]
+        (resp/response updated)))))
+
+(defn update-review [db topic-type topic-id review-status review-comment email]
+  (let [topic-name (cond
+                     (contains? (set constants/resource-types) topic-type) "resource"
+                     (= topic-type "project") "initiative"
+                     :else topic-type)
+        current-user (db.stakeholder/stakeholder-by-email (:spec db) {:email email})]
+    (jdbc/with-db-transaction [conn (:spec db)]
+      (let [existing (db.review/review-by-topic-item
+                      conn
+                      {:topic-name topic-name :topic-id topic-id})
+            review-id (:id existing)]
+        (if (or (nil? review-id)
+                ;; Assigned to another reviewer
+                (not (= (:reviewer existing) (:id current-user))))
+          {:status 403 :body {:message "Cannot update review for this topic"}}
+          (resp/response (db.review/update-review
+                          conn
+                          {:id review-id
+                           :review-status review-status
+                           :review-comment review-comment})))))))
+
+(defmethod ig/init-key ::get-reviewers [_ {:keys [db]}]
   (fn [_]
     (get-reviewers db)))
+
+(defmethod ig/init-key ::get-review [_ {:keys [db]}]
+  (fn [{{{:keys [topic-type topic-id]} :path} :parameters}]
+    (get-review db topic-type topic-id)))
```

Comment:
why we don't include de implementation inside these defmethods instead of extracting the logic to a new function?

Reply:
It seems easier to (re) evaluate the function, when developing and modifying them. We'd otherwise have to `reset` the system, when we make changes to these functions. We could move the code into the `defmethod`s all across the codebase, if we think that's a better way to define them.